dicom_numpy/combine_slices.py

```diff
@@ -75,27 +75,27 @@ def combine_slices(slice_datasets, rescale=None):
 
 
 def _merge_slice_pixel_arrays(slice_datasets, rescale=None):
-    first_dataset = slice_datasets[0]
-    num_rows = first_dataset.Rows
-    num_columns = first_dataset.Columns
-    num_slices = len(slice_datasets)
-
     sorted_slice_datasets = _sort_by_slice_position(slice_datasets)
 
     if rescale is None:
         rescale = any(_requires_rescaling(d) for d in sorted_slice_datasets)
 
-    if rescale:
-        voxels = np.empty((num_columns, num_rows, num_slices), dtype=np.float32, order='F')
-        for k, dataset in enumerate(sorted_slice_datasets):
+    first_dataset = sorted_slice_datasets[0]
```

Comment:
I won't have a chance to read through this in detail until later, but two comments:

1. We should add a test
2. What happens if some slices are RGB and others aren't? Would the error message make the problem clear to a user?

Reply:
BTW thanks for making a PR for this---it will definitely make dicom-numpy more useful!

Reply:
My pleasure :)

1. Yes, will add a bit later
2. I do not think it is a legitimate situation (?). My guess is that along with `Rows`, `Columns`, `BitsStored` - the attribute `SamplesPerPixel` is necessary for calculating `pixel_array` (and its shape) from `PixelData`. If there are some slices with different shape, they cannot be combined.
 The current error will be something related to numpy's incompatible shapes.
 In order to avoid it, maybe we should add `SamplesPerPixel` to the `invariant_properties` list? (in `_validate_slices_form_uniform_grid`)